src/CppUTest/Utest.cpp

```diff
@@ -575,6 +583,18 @@ UtestShell* UtestShell::getCurrent()
     return currentTest_;
 }
 
+const TestTerminator &UtestShell::getCurrentTestTerminator()
+{
+    return *currentTestTerminator_;
+}
+
+void UtestShell::setCrashOnFail(bool crashOnFail)
+{
+    if (crashOnFail)
+        currentTestTerminator_ = &crashingTestTerminator;
+    else
+        currentTestTerminator_ = &normalTestTerminator;
+}
```

Comment:
Can we split this into two functions (set and restoreDefault) so that we don't have this boolean and if.